api/controllers/userController.js

```diff
@@ -1,31 +1,62 @@
+const bcrypt = require('bcryptjs')
+const User = require('../schemas/userSchema')
 const userModel = require('../models/userModel')
 const passValidator = require('../../helpers/utility')
+const auth = require('./authController')
 
 module.exports = user = {
 
-    checkExistingUser: function(req,res,next){
-        userModel.checkExistingUser(req.body,function(error,result){
+    checkExistingUser: function (req, res, next) {
+        userModel.checkExistingUser(req.body, function (error, result) {
             // console.log(error)
-            if(error){
+            if (error) {
                 return res.status(500).send(error)
             }
-            else if(result){
+            else if (result) {
                 return res.status(400).send(result)
             }
             next()
         })
     },
 
-    createNewUser: function(req,res){
-        if(!passValidator(req.body.password))return res.status(400).send('Password must contain 1 numeric and 1 special character')
-        userModel.newUser(req.body, function(error,result){
-            if(error){
-                return res.status(400).send({error})
+    createNewUser: function (req, res) {
+        if (!passValidator(req.body.password)) return res.status(400).send('Password must contain 1 numeric and 1 special character')
+        userModel.newUser(req.body, function (error, result) {
+            if (error) {
+                return res.status(400).send({ error })
             }
-            if(!result){
+            if (!result) {
                 return res.status(400).send('something went wrong')
             }
             res.status(201).send('Success!')
         })
+    },
+
+    validateCredential: async function (req, res) {
+        const email = req.body.email
```

Comment:
@CharuTamar try to use let instead of const

use const only for global variables